include/smac_planner/smac_planner_hybrid.hpp

```diff
@@ -59,8 +61,34 @@ class SmacPlannerHybrid : public mbf_costmap_core::CostmapPlanner
     std::string name,
     costmap_2d::Costmap2DROS* costmap_ros) override;
 
+    struct PlanResult {
+      uint32_t result_code = mbf_msgs::GetPathResult::SUCCESS;
+      double cost = 0;
+      std::vector<geometry_msgs::PoseStamped> path{};
+      std::string message = "";
+      double length = 0;
+
+      bool isValid() const
+      {
+        return result_code == mbf_msgs::GetPathResult::SUCCESS;
+      }
+
+
+      PlanResult operator+(const PlanResult& other_result) const{
+        if (!isValid() || !other_result.isValid()){
+          return PlanResult{mbf_msgs::GetPathResult::FAILURE, 0.0, {}, "One of the segments is invalid, cannot add paths", 0};
+        }
+
+        PlanResult combined = *this;
+        combined.cost += other_result.cost;
+        combined.length += other_result.length;
+        combined.path.insert(combined.path.end(), other_result.path.begin() + 1, other_result.path.end());
```

Comment:
F: here we remove the first point, which works for our case, since we assume the end of one result is the start of the next one, but maybe not true for a general case.
Need a check to verify if the assumption is true